account_credit_control_attach_invoice/models/report.py

```diff
@@ -12,8 +12,15 @@ class IrActionsReport(models.Model):
     _inherit = "ir.actions.report"
 
     def _render_qweb_pdf(self, report_ref, res_ids=None, data=None):
-        reports = ["account_credit_control.report_credit_control_summary"]
-        if report_ref in reports and self.env.user.company_id.report_to_attach_id:
+        if isinstance(report_ref, models.BaseModel):
+            report_xml_id = report_ref.xml_id
+        else:
+            report_xml_id = report_ref
```

Comment:
The safest way to deal with the fact that `report_ref` can be many things, is to call `_get_report`:
https://github.com/odoo/odoo/blob/d33d10e6297d6e8dc9521cdae86e5bd3e8d7d55b/odoo/addons/base/models/ir_actions_report.py/#L647-L655

So, something like this:

```suggestion
        report = self._get_report(report_ref)
        report_xml_id = report.xml_id
```

Reply:
You are the best @ivantodorovich thanks for the tip !!